index.js

```diff
@@ -99,12 +99,32 @@ module.exports = {
 
     // We shouldn't enforce filename casing to always be the same.  e.g. We may
     // prefer kebab-case for bin files, PascalCase for React components, etc.
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/filename-case.md
     'unicorn/filename-case': 'off',
 
     // Uses safe-regex to disallow potentially catastrophic exponential-time
     // regular expressions.
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-unsafe-regex.md
     'unicorn/no-unsafe-regex': 'error',
 
+    // This rule is onerous in React functional components, has been debated
+    // by the community and proven as an unnecessary optimization
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/consistent-function-scoping.md
+    'unicorn/consistent-function-scoping': 'off',
+
+    // This rule is incompatible with React
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-null.md
+    'unicorn/no-null': 'off',
+
+    // Preventing abbreviations is too opinionated for linting and is not
```

Comment:
"too opinionated" is probably itself an opinion, what's the intent behind this phrase?

Reply:
This rule is too nit-picky. It's too micromanag-y. It's too idealistic.

If you don't like abbreviations don't use them. But there are plenty of valid cases for them. In React, especially.

If you don't disable this rule, I will. And so will a lot of other people. So save us all some trouble and leave it disabled.

Reply:
Here's one example of why this rule is unrealistic. It's incompatible with the no-shadow rule.

I may have a legitimate use for the variable name "index" in my code. This rule requires I use the word "index" and not "i" in array functions. So, now I'm faced with the decision to change the name of my "index" variable, or I have to think of a different variable name to use for "i" even though "i" is something all Javascript developers are familiar with.

Naming things is hard. Sometimes abbreviations are ok. Let me decide. Not all code needs to be written like the iPhone API.